crates/partition-store/tests/outbox_table_test/mod.rs

```diff
@@ -31,21 +30,37 @@ pub(crate) async fn populate_data<T: OutboxTable>(txn: &mut T) {
         .await;
 
     // add a successor and a predecessor partitions
-    txn.add_message(partition1336, 0, mock_outbox_message())
+    txn.add_message(PartitionId::from(1336), 0, mock_outbox_message())
         .await;
     txn.add_message(PartitionId::from(1338), 0, mock_outbox_message())
         .await;
 }
 
 pub(crate) async fn consume_message_and_truncate<T: OutboxTable>(txn: &mut T) {
     let partition1337 = PartitionId::from(1337);
-    let mut sequence = 0;
-    while let Ok(Some((seq, _))) = txn.get_next_outbox_message(partition1337, sequence).await {
-        sequence = seq + 1;
+    let mut count = 0;
+    let mut max_seq_id = 0;
+    while let Ok(Some((seq, _))) = txn.get_next_outbox_message(partition1337, count).await {
+        count += 1;
+        max_seq_id = seq;
     }
-    assert_eq!(sequence, 4);
+    assert_eq!(count, 4);
+
+    // truncate range
+    txn.truncate_outbox(partition1337, 0..=max_seq_id - 1).await;
 
-    txn.truncate_outbox(partition1337, 0..sequence).await;
+    let result = txn
+        .get_next_outbox_message(partition1337, 0)
+        .await
+        .expect("should not fail");
+    assert!(
+        result.is_some(),
+        "one message should remain in outbox after the first truncation"
+    );
```

Comment:
I realized that we're not testing the scenario where we want to truncate a range, but keep some messages behind. There was an off-by-one error in test code since it was deleting up to sequence_id `4` rather than '3' - hence the explicit split between outbox message count and the max sequence id encountered.

Reply:
I know that the `partition-store` tests are a bit rigid. But instead of making the existing tests more complicated by testing some aspects as part of some bigger tests, I would suggest creating a dedicated test for the case you want to cover (e.g. truncating a single message).